pydantic/decorators.py

```diff
@@ -116,21 +117,51 @@ def __call__(self, __cls: Any, __values: _decorators.RootValidatorValues) -> _de
 )
 
 
+# Use a single overload here to get a different signature that requires at least one field be specified
```

Comment:
Why don’t we do the same thing just without an overload? Or at least using TYPE_CHECKING? A single overload is wrong

Reply:
> A single overload is wrong

I did this for two main reasons:

* I prefer the IDE behavior of a single overload, because jump-to-definition takes me to the implementation, rather than the `@overload` type signature. With `if TYPE_CHECKING:`, it will take me to the type-checking signature rather than the implementation, which I usually don't want (cough cough ConfigDict, 😩).

* I wanted it to be visually similar to the `@field_validator` decorator. In that case, because there are currently two signatures, the `@overload` is allowed. I'll note that even there, in principle we don't need two `@overload`s — with 
  ```python
  _V2ValidatorType = TypeVar(
      '_V2ValidatorType', bound=Union[V2WrapValidator, V2Validator, 'classmethod[Any]', 'staticmethod[Any]']
  )
  ```
  it could be expressed with one:
  ```python
  def field_validator(...) -> Callable[[_V2ValidatorType], _V2ValidatorType]:
  ```
  That's not to say that I'd suggest we change it — I prefer that it use @overload because it lets us have the different (compatible) signature in the implementation without resorting to the (in my opinion) less ergonomic `if TYPE_CHECKING:`.

-----

At any rate, I am happy using any of the following approaches:

* Single `@overload` (current approach)
* Changing the single `@overload` to `if TYPE_CHECKING:`
* Just dropping the `@overload` signature (and losing the mypy error for not specifying even one field name)
* Changing the implementation signature to `(__field, *fields, ...` and relying on the missing-positional-argument TypeError produced by python, rather than the more explicit error message.

Just let me know which you prefer.

Reply:
> Changing the implementation signature to (__field, *fields, ... and relying on the missing-positional-argument TypeError produced by python, rather than the more explicit error message.

I prefer this version. You’ll also have to check that __field is not a function if you want to give a runtime error for calling with no args

Reply:
Also, we definitely need the overload on field_validator to enforce the relationship between mode and the function signature

Reply:
Missed that, makes sense

Reply:
I've changed it back to using `__field, *fields` in the implementation and dropped the single overload.